### PR TITLE
ci: implement docbot in cyborg

### LINKED ISSUE #3855: Process docs updates
### What type of enhancement is this?

Other

### Background

When prototyping #3849 and developing the cyborg automation in #3852 and #3854, I found the following shortcomings in our current docs update process:

1. It creates an issue on the docs repo when a label is created, but a label created by actions cannot further trigger the create issue action [1]. For example, there is no corresponding issue for https://github.com/GreptimeTeam/greptimedb/pull/3823.
2. It cannot identify if there are duplicate issues already.

[1] This is by design from GitHub Actions to prevent endless recursive event trigger.

Generally, automation add labels and users read the labels to take actions. Or Users add labels and trigger automation based on the label: but never mixed.

### Proposal

I'd first hear from @sunng87 and @nicecui for how you currently handle cloud follow-ups and docs follow-ups and thus establish a clean but still solid way to handle these cases.

My current opinion is:

1. Remove the cloud follow-up label. It is rare and we can simply create an issue in the cloud repo and refer to the related issues/prs.

2. Replace the docs updates label with:
  * docs-todo
  * docs-done

We don't need a label for docs no-need. But we add a `docs-todo` label on `- [ ]  This PR does not require documentation updates.`. Then @nicecui and @fengjiachun can ensure that all the necessary docs are written by moving all the `docs-todo` label to `docs-done`. Whether or not create an issue on the docs repo is an implementation details.

P.S. This is also how CockrochDB handles its docs process.

Looking forward to your thoughts. cc @killme2008

### PR DESCRIPTION
**Pull request by tisonkun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

This closes #3855. Let me test it out in the fork first.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.
- [x]  This PR does not require documentation updates.

### CODE REVIEW

**Comment by tisonkun** [comment]
A few improvements can be made. Testing at https://github.com/tisonkun/greptimedb/actions/runs/9076211353/job/24938429237?pr=4.

**Comment by tisonkun** [comment]
Should be ready for review now.

File: cyborg/bin/follow-up-docs-issue.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        })
+        await client.rest.issues.addLabels({
+            owner, repo, issue_number: number, labels: [labelDocsRequired],
+        })
+        await docsClient.rest.issues.create({
+            owner: 'GreptimeTeam',
+            repo: 'docs',
+            title: `Update docs for ${title}`,
+            body: `A document change request is generated from ${html_url}`,
+            assignee: actor,
```
Comment (evenyag) [inline cyborg/bin/follow-up-docs-issue.ts:63]: Does the actor always have permission to create an issue in the docs repo?

Comment (tisonkun) [reply cyborg/bin/follow-up-docs-issue.ts:63]: You may find `const docsClient = obtainClient("DOCS_REPO_TOKEN")` and the token was previously configured to be able to create issues in the docs repo. Specifically, it's @sunng87's fine-grained PAT.

**Review by evenyag — APPROVED** [review]

**Comment by tisonkun** [comment]
cc @nicecui may you check this workflow makes sense to you.

**Review by nicecui — APPROVED** [review]
LGTM

**Pull request merged**